core/src/main/java/org/infinispan/marshall/persistence/impl/PersistenceMarshallerImpl.java

```diff
@@ -82,39 +81,27 @@ public void start() {
          userMarshaller.start();
```

Comment:
I know I came in a little later here, but I noticed this returns `this` in `getUserMarshaller` however `mediaType()` always returns `APPLICATION_PROTOSTREAM` is that correct? That sounds like the user marshaller only can support protostream?

Reply:
The bytes returned by `PersistenceMarshallerImpl` will always be of type `APPLICATION_PROTOSTREAM` regardless of whether a user marshaller is configured or not. This is because objects marshalled via a configured user marshaller are wrapped in a `MarshallableUserObject` instance which is always marshalled via Protostream.

Reply:
Gotcha